src/video_core/cache_storage.h

```diff
@@ -1,20 +1,23 @@
 // SPDX-FileCopyrightText: Copyright 2025 shadPS4 Emulator Project
 // SPDX-License-Identifier: GPL-2.0-or-later
 
+#pragma once
+
 #include "common/path_util.h"
 #include "common/singleton.h"
 #include "common/types.h"
 
 #include <functional>
+#include <thread>
 #include <vector>
 
-namespace Vulkan {
 namespace Storage {
 
 enum class BlobType : u32 {
     ShaderMeta,
     ShaderBinary,
     PipelineKey,
+    ShaderProfile,
 };
 
 class DataBase {
```

Comment:
Does this class have to be a singleton? Seems to be mostly used from PipelineCache methods, so could it just be a member of that class and pass just a reference to any static functions in vk_pipeline_serialization that need it